sampledkd/training/entrypoint_utils.py

```
@@ -0,0 +1,432 @@
+    devmap = getattr(teacher, "hf_device_map", None)
+    if isinstance(devmap, dict):
+        bad = [k for k, v in devmap.items() if isinstance(v, str) and ("cpu" in v or "disk" in v)]
+        if bad:
+            raise RuntimeError(f"Unexpected CPU/offloaded modules in device_map: {bad}")
+
+    print(f"[teacher] 8-bit sharded. Device map: {devmap}", flush=True)
+    # Use the first teacher GPU for feeding inputs
+    return teacher, tok, torch.device(f"cuda:{g0}")
+
+
+def load_teacher_8bit_strict_2gpus(
```

Ruff (0.13.3)

Redefinition of unused `load_teacher_8bit_strict_2gpus` from line 103

(F811)